src/output/scanline_output_file.rs

```diff
@@ -163,20 +172,34 @@ impl<'a> ScanlineOutputFile<'a> {
         }
     }
 
-    /// Writes image data from the given FrameBuffer.
+    /// Writes the image incrementally over multiple calls.
+    ///
+    /// `framebuffer` may have a different vertical resolution than the image,
+    /// but it must have the same horizontal resolution.  Multiple calls will
+    /// incrementally write chunks of scanlines in the order given until the
+    /// image is complete.
+    ///
+    /// For example, to write a 2000-pixel-tall image, you could call this
+    /// function four times with 500-pixel-tall FrameBuffers.
     ///
-    /// The passed FrameBuffer may have a different vertical resolution than the
-    /// image, but must have the same horizontal resolution.  Multiple calls to
-    /// this method in sequence will incrementally write subsequent vertical
-    /// chunks of the image.
+    /// If `framebuffer` has more scanlines than remain in the image, only the
+    /// remaining scanlines will be written.
     ///
-    /// If the FrameBuffer has fewer scanlines than the remaining scanlines in
-    /// the image, then only that many scanlines will be written.  If the
-    /// FrameBuffer has more scanlines than remain in the image, then the
-    /// only the remaining number of scanlines will be written from the
-    /// FrameBuffer, and the image will be complete.
+    /// Note: all scanlines must be written for the resulting OpenEXR file to
+    /// be complete and correct.
     ///
     /// On success returns the number of scanlines written.
+    ///
+    /// # Errors
+    ///
+    /// This function expects `framebuffer` to have the same _horizontal_
+    /// resolution as the output file, as well as the same channels (with
+    /// matching types and subsampling).
+    ///
+    /// It will also return an error if:
+    ///
+    /// * All scanlines of the image have already been written.
```

Comment:
Discussed on IRC and decided to change to rejecting FrameBuffers that have more than the remaining scanlines.